sympy/interactive/printing.py

```diff
@@ -300,8 +292,7 @@ def _result_display(self, arg):
             debug("init_printing: using mathjax formatter")
             for cls in printable_types:
                 latex_formatter.for_type(cls, _print_latex_text)
-            for typ in sympy_latex_types:
-                typ._repr_latex_ = typ._repr_latex_orig
+            Printable._repr_latex_ = Printable._repr_latex_orig
```

Comment:
Perhaps this logic should be moved to the class (and we can probably do the same for png).

Reply:
I think it should be doable with `__getattr__`, unless there's some technical detail about it I'm missing.

Reply:
We need access to the local `settings` and `latex_printer` variables in the implementation of `_repr_latex_`. While there may be a nicer way to handle it, I think it's a little involved, and I'd rather leave it to a follow-up patch.